repository/src/main/java/org/pentaho/platform/repository2/unified/ExceptionLoggingDecorator.java

```diff
@@ -89,11 +93,17 @@ public Boolean call() throws Exception {
 
   public RepositoryFile createFile( final Serializable parentFolderId, final RepositoryFile file,
       final IRepositoryFileData data, final String versionMessage ) {
+    Constructor exceptionConstructor = null;
+    try {
+      exceptionConstructor = UnifiedRepositoryCreateFileException.class.getConstructor( String.class, Throwable.class );
+    } catch ( NoSuchMethodException e ) {
+      e.printStackTrace();
```

Comment:
Pretty sure we shouldn't have printStackTrace() in exception handlers. Especially since it's swallowing the exception at that point. If you do get that exception, you will end up with an NPE below. Please fix this.